### PR TITLE
fix(agg-state): use correct distribution indices

### LINKED ISSUE #4130: bug: dist key indices infer not correct.
Currently we use input dist key indices directly for agg state table. This is not correct, cuz the dist key indices is on input chunk.
https://github.com/singularity-data/risingwave/blob/093addfaf4cf6c728cc8c20c15a7aa2616617794/src/frontend/src/optimizer/plan_node/logical_agg.rs#L306

However, the state table needs dist key indices on internal table columns.

We need `column_mapping` in #4108 to do the transformation: convert input dist key indices -> state table dist key indices

cc @st1page

### PR DESCRIPTION
**Pull request by BowenXiao1999**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Close #4130 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by BowenXiao1999** [comment]
cc @likg227 may test this to see whether fix the problem? 😢

**Comment by BowenXiao1999** [comment]
Well, seems some bug. Convert to draft and keep looking into it.

**Comment by BugenZhao** [comment]
Do we have any e2e tests to reflect this bug?

**Comment by BowenXiao1999** [comment]
> Do we have any e2e tests to reflect this bug?

I guess no. Otherwise previous pr won't get merged. But indeed looks a potential problem and some people encounter seems encounter it in new feature

**Comment by likg227** [comment]
> Do we have any e2e tests to reflect this bug?

distinct agg for stream can reflect this bug, but there are other blocking bugs.

File: src/frontend/src/optimizer/plan_node/logical_agg.rs [hunk 1]
```diff
@@ -334,16 +334,20 @@ impl LogicalAgg {
...
-        let get_value_state_table = |value_key: usize| -> TableCatalog {
+        let get_value_state_table = |value_key_out: usize| -> TableCatalog {
             let mut internal_table_catalog_builder = TableCatalogBuilder::new();
             for &idx in &self.group_key {
                 let column_idx = internal_table_catalog_builder.add_column(&in_fields[idx]);
                 internal_table_catalog_builder.add_order_column(column_idx, OrderType::Ascending);
             }
-            internal_table_catalog_builder.add_column(&out_fields[value_key]);
+            internal_table_catalog_builder.add_column(&out_fields[value_key_out]);
             internal_table_catalog_builder.build(in_dist_key.clone(), in_append_only)
```
Comment (BowenXiao1999) [inline src/frontend/src/optimizer/plan_node/logical_agg.rs:351]: I was tring to do the fix for value state, but some input dist key can not always been found in table columns

**Review by st1page — APPROVED** [review]
LGTM

**Comment by BowenXiao1999** [comment]
Test locally that distinct agg panic is resolved. Let's merge

**Pull request merged**